In [1]:
%whos

Interactive namespace is empty.


In [1]:
properties = pd.read_csv("properties_2017.csv")

C:\ProgramData\Anaconda3\lib\site-packages\IPython\core\interactiveshell.py:2728: DtypeWarning: Columns (49) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [30]:
len_properties = len(properties)

In [2]:
properties.describe().iloc[:, 20:39].T

count          mean            std           min  \
garagecarcnt             891008.0  1.826581e+00       0.608934  0.000000e+00   
garagetotalsqft          891008.0  3.831562e+02     246.219420  0.000000e+00   
heatingorsystemtypeid   1869164.0  4.083750e+00       3.286780  1.000000e+00   
latitude                2982285.0  3.400131e+07  243515.706879  3.332439e+07   
longitude               2982285.0 -1.182017e+08  345591.772190 -1.194758e+08   
lotsizesquarefeet       2712511.0  2.260376e+04  249983.626731  1.000000e+02   
poolcnt                  539632.0  1.000000e+00       0.000000  1.000000e+00   
poolsizesum               27958.0  5.197154e+02     191.325869  1.900000e+01   
pooltypeid10              17006.0  1.000000e+00       0.000000  1.000000e+00   
pooltypeid2               33056.0  1.000000e+00       0.000000  1.000000e+00   
pooltypeid7              505895.0  1.000000e+00       0.000000  1.000000e+00   
propertylandusetypeid   2982285.0  2.600508e+02      15.931208  3.100000e+01   
rawcensustractandblock  2982285.0  6.048367e+07  200891.402459  6.037101e+07   
regionidcity            2923089.0  3.498766e+04   50709.676616  3.491000e+03   
regionidcounty          2982285.0  2.569085e+03     788.683964  1.286000e+03   
regionidneighborhood    1156741.0  1.935387e+05  165725.266428  6.952000e+03   
regionidzip             2972503.0  9.655329e+04    3680.815407  9.598200e+04   
roomcnt                 2982248.0  1.474357e+00       2.839869  0.000000e+00   
storytypeid                1623.0  7.000000e+00       0.000000  7.000000e+00   

                                 25%           50%           75%           max  
garagecarcnt            2.000000e+00  2.000000e+00  2.000000e+00  2.500000e+01  
garagetotalsqft         3.120000e+02  4.410000e+02  4.940000e+02  7.749000e+03  
heatingorsystemtypeid   2.000000e+00  2.000000e+00  7.000000e+00  2.400000e+01  
latitude                3.382740e+07  3.400807e+07  3.416186e+07  3.481965e+07  
longitude              -1.183929e+08 -1.181721e+08 -1.179489e+08 -1.175543e+08  
lotsizesquarefeet       5.683000e+03  7.000000e+03  9.893000e+03  3.710005e+08  
poolcnt                 1.000000e+00  1.000000e+00  1.000000e+00  1.000000e+00  
poolsizesum             4.300000e+02  4.950000e+02  5.940000e+02  1.741000e+04  
pooltypeid10            1.000000e+00  1.000000e+00  1.000000e+00  1.000000e+00  
pooltypeid2             1.000000e+00  1.000000e+00  1.000000e+00  1.000000e+00  
pooltypeid7             1.000000e+00  1.000000e+00  1.000000e+00  1.000000e+00  
propertylandusetypeid   2.610000e+02  2.610000e+02  2.610000e+02  2.790000e+02  
rawcensustractandblock  6.037400e+07  6.037571e+07  6.059042e+07  6.111009e+07  
regionidcity            1.244700e+04  2.521800e+04  4.545700e+04  3.965560e+05  
regionidcounty          1.286000e+03  3.101000e+03  3.101000e+03  3.101000e+03  
regionidneighborhood    4.673600e+04  1.189200e+05  2.748000e+05  7.641670e+05  
regionidzip             9.618000e+04  9.637700e+04  9.697400e+04  3.996750e+05  
roomcnt                 0.000000e+00  0.000000e+00  0.000000e+00  9.600000e+01  
storytypeid             7.000000e+00  7.000000e+00  7.000000e+00  7.000000e+00

In [6]:
properties.keys(), len(properties.columns)

(Index(['parcelid', 'airconditioningtypeid', 'architecturalstyletypeid',
        'basementsqft', 'bathroomcnt', 'bedroomcnt', 'buildingclasstypeid',
        'buildingqualitytypeid', 'calculatedbathnbr', 'decktypeid',
        'finishedfloor1squarefeet', 'calculatedfinishedsquarefeet',
        'finishedsquarefeet12', 'finishedsquarefeet13', 'finishedsquarefeet15',
        'finishedsquarefeet50', 'finishedsquarefeet6', 'fips', 'fireplacecnt',
        'fullbathcnt', 'garagecarcnt', 'garagetotalsqft', 'hashottuborspa',
        'heatingorsystemtypeid', 'latitude', 'longitude', 'lotsizesquarefeet',
        'poolcnt', 'poolsizesum', 'pooltypeid10', 'pooltypeid2', 'pooltypeid7',
        'propertycountylandusecode', 'propertylandusetypeid',
        'propertyzoningdesc', 'rawcensustractandblock', 'regionidcity',
        'regionidcounty', 'regionidneighborhood', 'regionidzip', 'roomcnt',
        'storytypeid', 'threequarterbathnbr', 'typeconstructiontypeid',
        'unitcnt', 'yardbuildingsqft17'

In [8]:
properties.index.tail(10)

AttributeError: 'RangeIndex' object has no attribute 'tail'

In [15]:
list(properties.columns).index('garagecarcnt'), list(properties.columns).index('regionidneighborhood')

(20, 38)

In [4]:
#properties.iloc[:, [20:39]].tail(10)
myprop = properties.iloc[:, 20:39]

In [20]:
properties.loc[0:10,['parcelid','garagecarcnt']]

parcelid  garagecarcnt
0   10754147           NaN
1   10759547           NaN
2   10843547           NaN
3   10859147           NaN
4   10879947           NaN
5   10898347           NaN
6   10933547           NaN
7   10940747           NaN
8   10954547           NaN
9   10976347           NaN
10  11070347           NaN

In [47]:
ind = properties["parcelid"]


AttributeError: 'Series' object has no attribute 'merge'

In [33]:
for i in range(len(myprop.columns)):
    print(myprop.columns[i], end='\n')

parcelid
garagecarcnt
garagetotalsqft
hashottuborspa
heatingorsystemtypeid
latitude
longitude
lotsizesquarefeet
poolcnt
poolsizesum
pooltypeid10
pooltypeid2
pooltypeid7
propertycountylandusecode
propertylandusetypeid
propertyzoningdesc
rawcensustractandblock
regionidcity
regionidcounty
regionidneighborhood


In [43]:
print(myprop.propertylandusetypeid.notnull())

0          False
1          False
2          False
3          False
4          False
5          False
6          False
7          False
8          False
9          False
10         False
11         False
12         False
13         False
14         False
15         False
16         False
17         False
18         False
19         False
20         False
21         False
22         False
23         False
24         False
25         False
26         False
27         False
28         False
29         False
           ...  
2985187    False
2985188    False
2985189    False
2985190    False
2985191    False
2985192    False
2985193    False
2985194    False
2985195    False
2985196    False
2985197    False
2985198    False
2985199    False
2985200    False
2985201    False
2985202    False
2985203    False
2985204    False
2985205    False
2985206    False
2985207    False
2985208    False
2985209    False
2985210    False
2985211    False
2985212    False
2985213    False
2985214    Fal

### propertyzoningdesc
#### 토지용도
- Property zoning description 
-  Description of the allowed land uses (zoning) for that property
- data type : str + str 
- type count : 5651
- length : 1982471
- null : 1002746

In [33]:
len(propdes)

1982471

In [22]:
propdes = properties.propertyzoningdesc[properties.propertyzoningdesc.notnull()]
propertyzoningdescCount =propdes.value_counts()

In [29]:
2985217 - len(propdes)

1002746

### rawcensustractandblock 
#### 위치정보
-  Census tract and block ID combined - also contains blockgroup assignment by extension
- 미 인구조사국의 GEOID 지리정보 ID. 정부에서 사용하는 국가통계, 법적, 인구추적용 ID인듯
- datatype : float
- digit : 7자리
- length : 2982285

### censustractandblock
-  Census tract and block ID combined - also contains blockgroup assignment by extension
- datatype :float
- digit: 13자리
- length: 2910232

In [43]:
properties.rawcensustractandblock.value_counts()

6.059052e+07    1576
6.037920e+07    1455
6.059052e+07     971
6.059063e+07     895
6.037137e+07     886
6.037577e+07     824
6.037920e+07     816
6.037274e+07     812
6.037277e+07     800
6.059063e+07     793
6.059052e+07     790
6.111007e+07     774
6.037920e+07     755
6.059052e+07     753
6.037208e+07     749
6.059053e+07     720
6.037142e+07     659
6.037275e+07     639
6.059022e+07     636
6.111008e+07     631
6.037236e+07     629
6.037262e+07     611
6.037261e+07     608
6.037544e+07     598
6.037113e+07     597
6.059075e+07     594
6.037140e+07     588
6.059074e+07     584
6.059032e+07     580
6.059032e+07     572
                ... 
6.037433e+07       1
6.037703e+07       1
6.037303e+07       1
6.037800e+07       1
6.037534e+07       1
6.037702e+07       1
6.037464e+07       1
6.037208e+07       1
6.037502e+07       1
6.037502e+07       1
6.037542e+07       1
6.037403e+07       1
6.037225e+07       1
6.037406e+07       1
6.037911e+07       1
6.037901e+07       1
6.037901e+07 

In [42]:
properties.censustractandblock.value_counts()

6.037920e+13    1171
6.037137e+13     881
6.037920e+13     811
6.059063e+13     807
6.059052e+13     806
6.037277e+13     800
6.037920e+13     779
6.111007e+13     771
6.037577e+13     752
6.059052e+13     740
6.037208e+13     718
6.059053e+13     685
6.037142e+13     650
6.111008e+13     631
6.037275e+13     630
6.037236e+13     628
6.059063e+13     613
6.059052e+13     613
6.059074e+13     609
6.037262e+13     608
6.037261e+13     599
6.037296e+13     596
6.059075e+13     593
6.037113e+13     590
6.037140e+13     586
6.059074e+13     584
6.059042e+13     582
6.059032e+13     580
6.037578e+13     569
6.059032e+13     568
                ... 
6.037911e+13       1
6.037901e+13       1
6.037530e+13       1
6.059001e+13       1
6.037209e+13       1
6.037104e+13       1
6.059086e+13       1
6.037911e+13       1
6.037600e+13       1
6.037225e+13       1
6.037311e+13       1
6.037901e+13       1
6.037536e+13       1
6.037403e+13       1
6.037462e+13       1
6.059063e+13       1
6.059012e+13 

### regionidcounty
#### 위치정보

- County in which the property is located
- length : 2982285
- datatype: float
- digit : 4자리
    - 3101
    - 1286
    - 2061
    
### regionidcity

-  City in which the property is located (if any)
- length: 2923089
- datatype: float
- digit : 최대 6자리
    - 186가지 regionid 

### regionidzip

-  Zip code in which the property is located
- datatype: float
- digit: 5자리
    - 403가지 zipcode

### regionidneighborhood

- Neighborhood in which the property is located
- datatype: float
- digit : 최대 6자리
    - 529가지 regionid

In [36]:
properties.regionidcounty.value_counts()

3101.0    2012741
1286.0     745800
2061.0     223744
Name: regionidcounty, dtype: int64

In [39]:
properties.regionidcity.value_counts()

12447.0     670868
46298.0      93454
16764.0      59793
52650.0      58619
25218.0      57013
54311.0      54302
5534.0       49123
47568.0      47763
40227.0      46279
13150.0      42112
34278.0      38791
27110.0      37820
45457.0      36090
54722.0      35618
24832.0      34448
47019.0      33806
33252.0      33336
34543.0      32552
12773.0      31381
24812.0      30356
20008.0      29232
14542.0      25781
53571.0      25035
51239.0      24814
396054.0     24589
24174.0      24166
12292.0      23034
21412.0      22562
32380.0      22465
33612.0      22267
             ...  
3491.0        1076
114834.0       798
6822.0         781
16389.0        760
13232.0        717
25271.0        696
34037.0        666
38980.0        623
42091.0        597
25621.0        582
33312.0        448
14906.0        425
118880.0       402
53162.0        274
32927.0        183
13311.0        182
10815.0        110
41484.0         86
21395.0         77
19202.0         24
48211.0         23
25535.0     

In [40]:
properties.regionidcity.count()

2923089

In [41]:
properties.regionidzip.value_counts()

96987.0     22025
96193.0     21756
97118.0     20615
97319.0     20337
96964.0     19923
97318.0     19880
96368.0     19774
96990.0     19215
96962.0     17813
97328.0     17322
97116.0     17169
97329.0     16583
96361.0     16557
97008.0     16209
96186.0     16051
97078.0     15798
96996.0     15776
96488.0     15676
96364.0     15580
96489.0     15528
96206.0     15467
96995.0     15282
97083.0     15245
96974.0     15162
96505.0     15123
96239.0     14996
96370.0     14721
96122.0     14709
97004.0     14682
96351.0     14361
            ...  
97324.0       647
95995.0       560
96148.0       531
95998.0       530
399675.0      433
97108.0       396
95991.0       379
96002.0       331
96226.0       305
96039.0       195
96329.0       100
97092.0        35
96467.0        35
97339.0        35
97177.0        19
96500.0        19
97111.0        10
97164.0         8
97088.0         5
97336.0         4
96114.0         2
97185.0         2
96069.0         2
96859.0         2
96953.0   

In [44]:
properties.regionidneighborhood.value_counts()

118208.0    32244
268496.0    23182
48570.0     21188
27080.0     18883
37739.0     18648
54300.0     17745
33183.0     16952
34213.0     16611
51906.0     15653
41131.0     15446
48200.0     14860
31817.0     14691
276476.0    14565
46736.0     14357
113455.0    14352
40548.0     13320
268588.0    12894
274049.0    12580
274514.0    12454
41466.0     11162
40215.0     10831
6952.0      10817
27987.0     10616
47880.0     10445
47950.0     10444
118920.0    10426
113910.0    10229
118872.0    10063
37835.0     10002
26134.0      9977
            ...  
764153.0       27
273042.0       26
763079.0       26
761214.0       25
761220.0       23
762176.0       21
762596.0       21
764139.0       20
761090.0       20
623346.0       18
764136.0       16
115295.0       14
761217.0       13
268446.0       11
761212.0       11
764137.0       10
260382.0       10
249692.0        9
764150.0        7
273622.0        7
762182.0        7
274108.0        7
764091.0        6
411726.0        5
764146.0  